src/envoy/tcp/tcp_cluster_rewrite/config.cc

```diff
@@ -0,0 +1,73 @@
+/* Copyright 2018 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "src/envoy/tcp/tcp_cluster_rewrite/config.h"
+#include "src/envoy/tcp/tcp_cluster_rewrite/tcp_cluster_rewrite.h"
+
+#include "envoy/registry/registry.h"
+#include "envoy/server/filter_config.h"
+#include "src/envoy/utils/config.h"
+
+using namespace ::istio::envoy::config::filter::network::tcp_cluster_rewrite;
+
+namespace Envoy {
+namespace Tcp {
+namespace TcpClusterRewrite {
+
+Network::FilterFactoryCb
+TcpClusterRewriteFilterConfigFactory::createFilterFactory(
+    const Json::Object& config_json, Server::Configuration::FactoryContext&) {
+  v2alpha1::TcpClusterRewrite config_pb;
+  if (!Utils::ReadV2Config(config_json, &config_pb) &&
+      !Utils::ReadV1Config(config_json, &config_pb)) {
+    throw EnvoyException("Failed to parse JSON config");
```

Comment:
I think the v1 is not needed. You can simply throw exception from this function.